deep_speech_2/decoder.py

```diff
@@ -36,25 +38,164 @@ def ctc_best_path_decode(probs_seq, vocabulary):
     return ''.join([vocabulary[index] for index in index_list])
 
 
-def ctc_decode(probs_seq, vocabulary, method):
+class Scorer(object):
     """
-    CTC-like sequence decoding from a sequence of likelihood probablilites. 
+    External defined scorer to evaluate a sentence in beam search
+               decoding, consisting of language model and word count.
 
-    :param probs_seq: 2-D list of probabilities over the vocabulary for each
-                      character. Each element is a list of float probabilities
-                      for one character.
-    :type probs_seq: list
+    :param alpha: Parameter associated with language model.
+    :type alpha: float
+    :param beta: Parameter associated with word count.
+    :type beta: float
+    :model_path: Path to load language model.
+    :type model_path: basestring
+    """
+
+    def __init__(self, alpha, beta, model_path):
+
+        self._alpha = alpha
+        self._beta = beta
+        self._language_model = kenlm.LanguageModel(model_path)
+
+    def language_model_score(self, sentence, bos=True, eos=False):
```

Comment:
The decoded prefix in ctc decoder doesn't contain any special tokens. So the reprocessing is simplified.